src/libraries/Microsoft.Extensions.Logging/src/LoggerMessageAttribute.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Extensions.Logging
+{
+    /// <summary>
+    /// Provides information to guide the production of a strongly-typed logging method.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method)]
+    public sealed class LoggerMessageAttribute : Attribute
+    {
+#pragma warning disable SA1629 // Documentation text should end with a period
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LoggerMessageAttribute"/> class
+        /// which is used to guide the production of a strongly-typed logging method.
+        /// </summary>
+        /// <remarks>
+        /// The method this attribute is applied to:
+        ///    - Must be a partial method.
+        ///    - Must return <c>void</c>.
+        ///    - Must not be generic.
+        ///    - Must have an <see cref="ILogger"/> as one of its parameters.
+        ///    - Must have a <see cref="Microsoft.Extensions.Logging.LogLevel"/> as one of its parameters.
+        ///    - None of the parameters can be generic.
+        /// </remarks>
+        /// <example>
+        /// static partial class Log
+        /// {
+        ///     [LoggerMessage(EventId = 0, Message = "Could not open socket for {hostName}")]
+        ///     static partial void CouldNotOpenSocket(ILogger logger, LogLevel level, string hostName);
+        /// }
+        /// </example>
+        public LoggerMessageAttribute() { }
+#pragma warning restore SA1629 // Documentation text should end with a period
+
+        /// <summary>
+        /// Gets the logging event id for the logging method.
+        /// </summary>
+        public int EventId { get; set; } = -1;
```

Comment:
```suggestion
        public int EventId { get; set; } = 0;
```
Could the default EventId be `0`? It may not be completely related to this issue but Serilog would not add `EventId` to its properties if it is `0`, but it will add for `-1`.
https://github.com/serilog/serilog-extensions-logging/blob/39307172d5116e430e974aed03dcb002571b3276/src/Serilog.Extensions.Logging/Extensions/Logging/SerilogLogger.cs#L157